src/scripts/farming/Farming.ts

```diff
@@ -2266,4 +2266,74 @@ class Farming implements Feature {
         }
     }
 
+    public handleWanderer(plot: Plot) {
+        if (!plot.canCatchWanderer()) {
+            return;
+        }
+        const wanderer = plot.wanderer;
+        const pokemonData = PokemonHelper.getPokemonByName(wanderer.name);
+        const berry = App.game.farming.berryData[plot.wanderer.berry];
+
+        const farmPoints = Math.floor(berry.farmValue / (4 + berry.growthTime[PlotStage.Bloom] / 1800));
+        App.game.wallet.gainFarmPoints(farmPoints);
+
+        const pokeball = App.game.pokeballs.calculatePokeballToUse(pokemonData.id, wanderer.shiny, false, EncounterType.wanderer);
+        if (pokeball !== GameConstants.Pokeball.None) {
+            wanderer.pokeball(pokeball);
+            wanderer.catching(true);
+            // Halved catch time in farm, it does not matter in the balance
+            setTimeout(() => this.attemptCatchWanderer(plot), App.game.pokeballs.calculateCatchTime(pokeball) / 2);
```

Comment:
Can we pass `pokeball` as a parameter to `attemptCatchWanderer` instead of having it as a property on the wanderer?

Reply:
We need the pokéball property for the catching animation anyway.

Reply:
Ah, ok, didn't see it was also used there